openedx/core/djangoapps/credentials_service/models.py

```diff
@@ -1,3 +1,270 @@
 """
 Models for the credentials service.
 """
+import os
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from model_utils.models import TimeStampedModel
+from xmodule_django.models import CourseKeyField
+
+
+MODE_CHOICES = (
+    (_('Honor'), 'honor'),
+    (_('Verified'), 'verified'),
+    (_('Professional'), 'professional'),
+)
+
+
+class AbstractCredential(TimeStampedModel):
+    """
+    Abstract Credentials configuration model to support multitenancy.
+    """
+    PROGRAMS, COURSES = (
+        "programs", "courses",
+    )
+    TYPES_CHOICES = (
+        (PROGRAMS, _("programs")),
+        (COURSES, _("courses")),
+    )
+
+    credential_type = models.CharField(max_length=32, choices=TYPES_CHOICES, default=COURSES)
+    site = models.ForeignKey('sites.Site', null=False, blank=False)
+    is_active = models.BooleanField(default=False)
+
+    def __unicode__(self):
+        """Unicode representation. """
+        return u"{site}, {credential_type}".format(
+            site=self.site,
+            credential_type=self.credential_type
+        )
+
+
+def assets_path(instance, filename):
+    """
+    Custom path for credentials templates and signatories file assets.
+
+    Arguments:
+        instance: CertificateTemplateAsset or Signatory object
+        filename: file to upload
+
+    Returns:
+        Path of asset file e.g.
+        credential_certificate_template_assets/1/filename,
+        signatories/1/filename
+    """
+
+    path = 'credential_certificate_template_assets'
+    if isinstance(instance, Signatory):
+        path = 'signatories'
+
+    name = os.path.join(
+        path,
+        str(instance.id),
+        filename
+    )
+    fullname = os.path.join(settings.MEDIA_ROOT, name)
+    if os.path.exists(fullname):
+        os.remove(fullname)
+
+    return name
+
+
+def validate_image(image):
```

Comment:
I don't see tests for either of the utility functions included in this module. Also, if these helpers are only used by a single model, I'd move them to the corresponding class.